src/installation/index.md

```diff
@@ -1,23 +1,57 @@
-  - `LLVM`: The Rust compiler uses `LLVM` as codegen backend.
+The `RISC-V` architecture has support in the mainline Rust compiler so setup is relatively simple, all we must do is add the appropriate compilation target.
+
+Hence, the dependencies required to develop Rust applications in `RISC-V` targets are:
+- Rust Toolchain with the proper target: Used to compile our Rust code.
+- `LLVM`: Used as codegen backend by the Rust compiler.
+- \[_Optional_\] `GCC` Toolchain: `GCC` linker can be used.
+  - `GCC` is marked as optional as you can also use `LLVM` linker.
+
+Aditonally when building `std` applications we also need:
+- [ESP-IDF]: Espressif IoT Development Framework.
+- [`ldproxy`] crate:  Simple tool to forward linker arguments given to [`ldproxy`] to the actual linker executable. The crate can be found in the [esp-rs/embuild] repository.
+
+## Rust in Xtensa targets
+To this day, there is no `Xtensa` support in the mainline Rust compiler, for this reason, we maintain the [esp-rs/rust] fork that adds support four our `Xtensa` targets.
 
-## `std` Applications
+`Xtensa` not being supported on Rust mainline is mainly a consequence of `LLVM` not supporting `Xtensa` targets. For that reason, we also maintain an LLVM fork with support for Espressif `Xtensa` targets in [espressif/llvm-project]
 
-- GCC Toolchain: GCC linker is used in our Rust applications.
+Another consequence of `LLVM` not suporting our `Xtensa` targets is that we need to provide our own linker. I.e. we'll need to install [GCC toolchain] to use it as our linker.
+
+> #### A note in upstreaming our forks.
+>
+> We are trying to upstream the changes in both our `LLVM` and Rust forks.
+> The first step is to upstream the `LLVM` project, this is already in progress
+> and you can see the status at this [tracking issue].
+> If our `LLVM` changes are accepted in `LLVM` mainline, we will proceed with trying
+> to upstream the Rust compiler changes.
+
+The forked compiler can coexist with the standard Rust compiler, so it is possible to have both installed on your system. The forked compiler is invoked when using the `esp` [channel] instead of the defaults, `stable` or `nightly`.
+
+Hence, the dependencies required to develop Rust applications in `Xtensa` targets are:
 - Rust Toolchain: Used to compile our Rust code.
-  - `LLVM`: The Rust compiler uses `LLVM` as codegen backend.
+  - _We need to use our custom fork with `Xtensa` support_.
+- `LLVM`: Used as codegen backend by the Rust compiler.
+  - _We need to use our custom fork with `Xtensa` support_.
+- `GCC` Toolchain: `GCC` linker is used in our Rust applications as it supports all our targets architectures (`Xtensa` and `RISC-V`).
+
+Aditonally when building `std` applications we also need:
 - [ESP-IDF]: Espressif IoT Development Framework.
 - [`ldproxy`] crate:  Simple tool to forward linker arguments given to [`ldproxy`] to the actual linker executable. The crate can be found in the [esp-rs/embuild] repository.
```

Comment:
This seems to be duplicated, as this exact text appeared previously on this page. Was this intentional? Just seems kind of odd at first glance.

Reply:
Alright, it's not a big deal so it's fine for now. Just wanted to make sure it wasn't an erroneous copy-paste.